### ISSUE TITLE
Possible solution for all dungeon/raid finder and arenas

### ISSUE THREAD

**Issue by kam92** [celguar/mangosbot-bots#96]
I'm sorry if it's already done this way, my research is not enough to find where this happens, but the logic for LFD/LFR and Arena can be instant for the player. A good fix would be to have whatever number of bots required and generated on the fly, as user request to use the group finder. Server chooses them from a list of builds (and not only the final builds, the talent points should be right for low level too, so the build path must be taken into consideration too, not just make final versions of level 80 builds) and go. 

For example, I'm a healer level 27, I queue for a dungeon on LFD. Server generates 4 characters between min and max dungeon level I queued for. Server checks how many tank,healer and dps needed, in this case needs 1 tank and 3 damage builds. random pick classes that has those builds and my temporary friends are ready. Player should also always be the leader of the group and in the end of the dungeon, if the players decides to to open dungeon finder and queue again, all bots should instantly accept the leader request. In the moment leader leaves the party, remaining members logout of the game and get deleted from the server database. 

I don't have much free time at the moment, but I have back-end development experience and I can code this myself, but if anyone has any information you think I should know before starting this, please share. Thank you.
